passthroughserver/main.cpp

```diff
@@ -23,64 +23,33 @@
 #include "waylandplatform.h"
 #endif // HAVE_WAYLAND
 #include "unknownplatform.h"
+#include "logging.h"
 
 #include <QGuiApplication>
 #include <QtDebug>
 
 namespace {
+QLoggingCategory::CategoryFilter defaultLoggingFilter;
 
-void disableMInputContextPlugin()
-{
-    // none is a special value for QT_IM_MODULE, which disables loading of any
-    // input method module in Qt 5.
-    setenv("QT_IM_MODULE", "none", true);
-}
-
-bool isDebugEnabled()
+void loggingCategoryFilter(QLoggingCategory *category)
 {
-    static int debugEnabled = -1;
-
-    if (debugEnabled == -1) {
+    if (qstrcmp(category->categoryName(), "driver.usb") == 0) {
         QByteArray debugEnvVar = qgetenv("MALIIT_DEBUG");
+        bool enabled = false;
         if (!debugEnvVar.isEmpty() && debugEnvVar != "0") {
-            debugEnabled = 1;
-        } else {
-            debugEnabled = 0;
+            enabled = true;
         }
+        category->setEnabled(QtDebugMsg, enabled);
+    } else if (defaultLoggingFilter) {
+        defaultLoggingFilter(category);
     }
-
-    return debugEnabled == 1;
 }
 
-void outputMessagesToStdErr(QtMsgType type,
-                            const QMessageLogContext &context,
-                            const QString &msg)
+void disableMInputContextPlugin()
```

Comment:
It seems this wasn't changed, but was removed and then added back in a different location, which makes the diff to `main.cpp` here quite a bit weird and extraneous. Could you perhaps move it back, and rebase so that the diff is cleaner and easier to understand without this unrelated change? Thanks.

Reply:
Actually I didn't touch it, but git considered the old log filtering to be similar enough to the new category filtering so it ended up showing delta with this moved instead. But since the function here is just wrapping one function call, let's just get rid of he extra indirection.